### ISSUE TITLE
Issue with typings

### ISSUE THREAD

**Issue by Renader** [TypeCtrl/ngx-emoji-mart#1]
Hello,

thanks for your effort so far. I've just added ngx-emoji-mart to my project and I receive some typing errors. I'm using it in a lazy loaded feature module of a almost fresh Angular CLI project. The errors prevent my app from building - but if I change another file the build is running through. There are no runtime errors in the browser - everything works fine expect my build/serve keeps throwing those errors. So it is probably only a issue with exported typings..

```
ERROR in node_modules/@ctrl/ngx-emoji-mart/anchors.component.d.ts(2,31): error TS2307: Cannot find module '@ctrl/ngx-emoji-mart/ngx-emoji'.
node_modules/@ctrl/ngx-emoji-mart/category.component.d.ts(2,37): error TS2307: Cannot find module '@ctrl/ngx-emoji-mart/ngx-emoji'.
node_modules/@ctrl/ngx-emoji-mart/emoji-frequently.service.d.ts(1,27): error TS2307: Cannot find module '@ctrl/ngx-emoji-mart/ngx-emoji'.
node_modules/@ctrl/ngx-emoji-mart/emoji-search.service.d.ts(1,41): error TS2307: Cannot find module '@ctrl/ngx-emoji-mart/ngx-emoji'.
node_modules/@ctrl/ngx-emoji-mart/picker.component.d.ts(2,50): error TS2307: Cannot find module '@ctrl/ngx-emoji-mart/ngx-emoji'.
node_modules/@ctrl/ngx-emoji-mart/preview.component.d.ts(2,41): error TS2307: Cannot find module '@ctrl/ngx-emoji-mart/ngx-emoji'.
node_modules/@ctrl/ngx-emoji-mart/skins.component.d.ts(2,23): error TS2307: Cannot find module '@ctrl/ngx-emoji-mart/ngx-emoji'.

```

My versions:
```
Angular CLI: 1.6.8
Node: 7.7.3
OS: linux x64
Angular: 5.2.4
... animations, common, compiler, compiler-cli, core, forms
... http, language-service, platform-browser
... platform-browser-dynamic, router

@angular/cdk: 5.2.0
@angular/cli: 1.6.8
@angular/material: 5.2.0
@angular-devkit/build-optimizer: 0.0.42
@angular-devkit/core: 0.0.29
@angular-devkit/schematics: 0.0.52
@ngtools/json-schema: 1.1.0
@ngtools/webpack: 1.9.8
@schematics/angular: 0.1.17
typescript: 2.5.3
webpack: 3.10.0

```

**Comment by scttcper** [comment]
I've only managed to get the warnings once and then they disappeared. Try out `0.0.4` I've reset the packages i'm using in the project to the stable versions. I also find trashing node_modules and reinstalling usually silences it. I think it has something to do with webpack's caching when adding it to a project. Not totally sure yet as I publish a few other packages in a similar way.

**Comment by scttcper** [comment]
okay with lazy loading i get the following which is from a completely different project than the one i'm testing with... wtf
```
ERROR in src/app/journey/journey-edit/journey-edit/journey-edit.component.html(109,11): : Property 'minDate' does not exist on type 'JourneyEditComponent'.
src/app/journey/journey-edit/journey-edit/journey-edit.component.html(110,11): : Property 'disableAll' does not exist on type 'JourneyEditComponent'.
src/app/journey/journey-edit/journey-edit/journey-edit.component.html(106,11): : Property 'calendarDate' does not exist on type 'JourneyEditComponent'.
src/app/journey/journey-edit/journey-edit/journey-edit.component.html(107,11): : Property 'pickDate' does not exist on type 'JourneyEditComponent'.
src/app/journey/journey-edit/journey-segment/journey-segment.component.html(17,57): : Property 'selectType' does not exist on type 'JourneySegmentComponent'. Did you mean 'selectedType'?
src/app/journey/journey-edit/journey-segment/journey-segment.component.html(24,72): : Property 'retrieveSegmentrLists' does not exist on type 'JourneySegmentComponent'.
```

**Comment by scttcper** [comment]
Lazy loading seems to be working as well https://github.com/TypeCtrl/emoji-mart-test

**Comment by Renader** [comment]
Wow that was a fast reply! I'll check as soon as I'm at my home pc.

**Comment by Renader** [comment]
The update did not helped. I checked out your demo repo and it worked. I digged deeper and could identify the typescript version as issue. 

It doesn't work with 0.0.4 and `"typescript": "2.5.3"` but it did work after upgrading to `typescript 2.6.2`. Since 2.5.x is still pretty common you might want to fix that - for me it works now since I have no issues with upgrading Typescript.

Thanks for your fast reaction!
